src/Deserializers/SnakDeserializer.php

```diff
@@ -107,6 +107,8 @@ private function newSomeValueSnak( array $serialization ) {
 
 	private function newValueSnak( array $serialization ) {
 		$this->requireAttribute( $serialization, 'datavalue' );
+		// Do not deserialize derived value snaks
+		$this->assertNotAttribute( $serialization, 'derived-values' );
```

Comment:
I did that exactly because we do it elsewhere: https://github.com/wmde/WikibaseDataModelSerialization/blob/master/src/Deserializers/TermDeserializer.php#L50